api/types/server.go

```diff
@@ -485,16 +501,39 @@ func (s *ServerV2) openSSHEC2InstanceConnectEndpointNodeCheckAndSetDefaults() er
 	return nil
 }
 
+// serverNameForEICE returns the deterministic Server's name for an EICE instance.
+// Returns an empty string if AccountID or InstanceID is not present.
+func serverNameForEICE(s *ServerV2) string {
+	awsAccountID := s.GetAWSAccountID()
+	awsInstanceID := s.GetAWSInstanceID()
+
+	if awsAccountID != "" && awsInstanceID != "" {
+		return fmt.Sprintf("%s-%s", awsAccountID, awsInstanceID)
+	}
+
+	return ""
+}
+
 // CheckAndSetDefaults checks and set default values for any missing fields.
 func (s *ServerV2) CheckAndSetDefaults() error {
 	// TODO(awly): default s.Metadata.Expiry if not set (use
 	// defaults.ServerAnnounceTTL).
 	s.setStaticFields()
 
-	// if the server is a registered OpenSSH node, allow the name to be
-	// randomly generated
-	if s.Metadata.Name == "" && s.IsOpenSSHNode() {
-		s.Metadata.Name = uuid.New().String()
+	if s.Metadata.Name == "" {
+		switch s.SubKind {
+		case SubKindOpenSSHEICENode:
+			// For EICE nodes, use a deterministic name.
+			eiceNodeName := serverNameForEICE(s)
+			if eiceNodeName == "" {
```

Comment:
Should we go heavy-duty here and use `api/utils/aws.IsEC2NodeID`?

Reply:
I'll add a test and a comment here